mcode/mcode-coverage-checker
# STEAM-849: Initial Patient Coverage Statistic Visualization

# Description
Issue: STEAM-849

Implements an initial patient coverage statistic visualization, laying the foundation for our initial visualization approach.

## (Feature) Demo/Screenshots
For convenience, I've included a screenshot to the new UI and the initial visualization. As you can see, this task also implemented the outcome coverage, since it came pretty cheaply. 
<img width="1116" alt="image" src="https://user-images.githubusercontent.com/5703736/162527427-fbbc027a-5c21-4dc6-9b04-1805f1e6e566.png">

## Important Changes

Note: I've changed around the directory structure a fair bit, as the number of files and components grows. I address the changes incrementally in the sections below.

`src/components/MainVisualization.js` and related `xViz.js` components
- Define a new visualization parent component, where we will add all future visualization components as we create them. 

`src/components/SimpleStatsViz.js`
- Generic component for our simplistic layout of statistics, styled by an accompanying CSS Module.

`src/data/fullBundle.json`
- In order to load the fullBundle in a visualization context, we need it to be in scope of the `src` directory. This local copy will need to be updated as we add more sections.

`src/lib/coverageChecker/coverageChecker.js`
- Defines the actual function used by the front-end, which calculates coverage reports for all the relevant sections.

`src/lib/coverageStats/coverageStats.js`
- Defines the functions for computing coverage statistics based on coverage reports

`src/lib/coverageStats/statsUtils.js`
- Defines utility functions for computing coverage stats

### More general changes

`MainSVG.js` and related files
- Migrated into an `oldSvg` subdirectory in the components directory; this can be removed gradually over time, but as we complete the other sections coverage checkers it might be useful to have the visualization local to our project still.

`src/pages/App.js`
- Changed to point at the MainVisualization instead of the SVG

`src/lib/logger.js` 
- Created a logger utility that we should be using instead of scattered console logs. If we ever decide to swap out to a more mature logger, we will only have to change things in one place

`README.md`
- Adds a meaningful README, inlining an example of Dylan M's coverage data model.

`.eslintrc`
- Added a temporary disabling of the prop-type requirement; as we mature our model for coverage data we should implement this code quality improvement. 

## Testing Recommendations
- Make sure that the stats make sense up (try removing a field or a resource and see how it effects the coverage calculation)
- Make sure that the functions I've created make sense.

# Checklists

**Submitter:**
- [x] Comment added to the relevant JIRA issue(s) with a link to this PR

@ACCT1 :
